File: problems/1574-maximum-product-of-two-elements-in-an-array/maximum-product-of-two-elements-in-an-array.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        # sortedNums = sorted(nums, reverse=True)
        # return (sortedNums[0] - 1) * (sortedNums[1] - 1)

        max1 = float('-inf')
        max2 = float('-inf')

        for n in nums:
            if n > max1:
                max2 = max1
                max1 = n
            elif n > max2:
                max2 = n
        return (max1 - 1) * (max2 -1 )
                

File: problems/1574-maximum-product-of-two-elements-in-an-array/maximum-product-of-two-elements-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const maxProduct = (nums) => {
    // Brute force
    // if (nums.length === 2) return (nums[0] - 1) * (nums[1] - 1);
    // let maxProduct = 0;
    // for (let i = 0; i < nums.length - 1; i++) {
    //     for (let j = i + 1; j < nums.length; j++) {
    //         const tmpProduct = (nums[i] - 1) * (nums[j] - 1);
    //         maxProduct = Math.max(maxProduct, tmpProduct);
    //     }
    // }
    // return maxProduct;

    const sortedNums = nums.sort((a, b) => b - a);
    return (sortedNums[0] - 1) * (sortedNums[1] - 1);
};